rs/core/corpora_cli/src/commands/infer.rs

```diff
@@ -0,0 +1,161 @@
+use crate::context::Context;
+use clap::Args;
+use corpora_client::models::{CorpusFileChatSchema, MessageSchema};
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::Path;
+use std::process::{Command, Stdio};
+
+const MAX_RETRIES: u8 = 1;
+
+#[derive(Args)]
+pub struct InferArgs {
+    #[arg(help = "Path to the file")]
+    pub path: String,
+
+    #[arg(short, long, help = "Check script to validate the output")]
+    pub check: Option<String>,
+}
+
+/// Read existing file content or return a default message
+fn get_current_file_content(path: &Path) -> String {
+    match fs::read_to_string(path) {
+        Ok(content) if !content.trim().is_empty() => content,
+        _ => "FILE CURRENTLY EMPTY. PLEASE INFER CONTENT.".to_string(),
+    }
+}
+
+/// The `infer` command: generates or refines a file
+pub fn run(ctx: &Context, args: InferArgs) {
+    let path = Path::new(&args.path);
+    let cwd = std::env::current_dir().expect("Failed to get current directory");
+    let absolute_path = cwd.join(path);
+    let relative_path = absolute_path
+        .strip_prefix(&ctx.corpora_config.root_path)
+        .expect("Failed to get relative path");
+
+    ctx.magenta(&format!("Inferring file: {}", relative_path.display()));
+    let ext = relative_path
+        .extension()
+        .and_then(|s| s.to_str())
+        .unwrap_or("");
+
+    // Read current file content if it exists
+    let current_file_content = get_current_file_content(path);
+
+    // Load metadata
+    let root_path = &ctx.corpora_config.root_path;
+    let voice = fs::read_to_string(root_path.join(".corpora/VOICE.md")).unwrap_or_default();
+    let purpose = fs::read_to_string(root_path.join(".corpora/PURPOSE.md")).unwrap_or_default();
+    let structure = fs::read_to_string(root_path.join(".corpora/STRUCTURE.md")).unwrap_or_default();
```

Comment:
Ultimately should probably just ... glob in as many different files and feed into the context k:v, filename:content and let some corpus direct the action how they think... I was feeling like I needed 2 distinct kinds of files for "structure" with the new `infer` command in the math textbook.